src/go/rpk/pkg/cli/cmd/acl/delete.go

```diff
@@ -12,378 +12,126 @@ package acl
 import (
 	"context"
 	"fmt"
-	"sort"
-	"strings"
 
-	"github.com/Shopify/sarama"
-	"github.com/pkg/errors"
-	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
-	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/ui"
-	"golang.org/x/sync/errgroup"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
 )
 
-func NewDeleteACLsCommand(
-	admin func() (sarama.ClusterAdmin, error),
-) *cobra.Command {
-	var (
-		resource        string
-		resourceName    string
-		namePattern     string
-		operations      []string
-		allowPrincipals []string
-		allowHosts      []string
-		denyPrincipals  []string
-		denyHosts       []string
-	)
-	command := &cobra.Command{
-		Use:          "delete",
-		Short:        "Delete ACLs",
-		SilenceUsage: true,
-		Args: func(_ *cobra.Command, _ []string) error {
-			if len(allowPrincipals) == 0 &&
-				len(denyPrincipals) == 0 {
-				return fmt.Errorf(
-					"at least one of --%s or --%s must be set",
-					allowPrincipalFlag,
-					denyPrincipalFlag,
-				)
+func NewDeleteCommand(fs afero.Fs) *cobra.Command {
+	var a acls
+	var printAllFilters bool
+	cmd := &cobra.Command{
+		Use:   "delete",
```

Comment:
Can delete command can have `dry-run` mode that prints ACLs that would be removed?

Can be done later.

Reply:
ACL deletion doesn't have a ValidateOnly field in the request itself, so there's no actual concept of a dry run for the request itself. One benefit of this PR is that (once the old behavior is removed), `rpk acl list` can be the dry run for `rpk acl delete`, since the flags and filters are identical. I plan to have a follow on PR greatly expanding the documentation of ACLs, and I plan to mention that.

Although, one option is to have a --dry-run _here_ just straight up call into the listing code.

Reply:
I've added `--dry` as an alias that calls into listing.